ocpmodels/datasets/ase_datasets.py

```diff
@@ -43,7 +50,7 @@ class AseAtomsDataset(Dataset):
 
     Derived classes must add at least two things:
         self.get_atoms_object(): a function that takes an identifier and returns a corresponding atoms object
-        self.id: a list of all possible identifiers that can be passed into self.get_atoms_object()
```

Comment:
Stylistic comment:
It would be better to add another abstract method called `def get_ids(self) -> List[int]`.
Then we would add a `self.ids = self.get_ids()` call here. That way, the derived classes can just implement the 2 abstract methods and not have to modify the constructor.

Reply:
Good idea! I added a function `load_dataset_get_ids(config)` to each class to handle all the custom loading and id list generation. None of the derived classes use a custom constructor now.